AcademyResidentInformationApi/V1/Gateways/AcademyGateway.cs

```diff
@@ -68,6 +68,52 @@ public ClaimantInformation GetClaimantById(int claimId, int personRef)
                 : MapPersonAndAddressesToClaimantInformation(databaseRecord.person, databaseRecord.address);
         }
 
+        public List<TaxPayerInformation> GetAllTaxPayers(string firstname = null, string lastname = null, string postcode = null, string address = null)
+        {
+            var firstNameSearchPattern = GetSearchPattern(firstname);
+            var lastNameSearchPattern = GetSearchPattern(lastname);
+            var addressSearchPattern = GetSearchPattern(address);
+            var postcodeSearchPattern = GetSearchPattern(postcode);
+
+            return (
+                from taxPayer in _academyContext.TaxPayers
+                join occupation in _academyContext.Occupations on taxPayer.AccountRef equals occupation.AccountRef
+                join property in _academyContext.CouncilProperties on occupation.PropertyRef equals property.PropertyRef
+                join email in _academyContext.Emails on taxPayer.AccountRef equals email.ReferenceId
+                join phoneNumber in _academyContext.PhoneNumbers on taxPayer.AccountRef.ToString() equals phoneNumber.Reference
+                where string.IsNullOrEmpty(firstname) || EF.Functions.ILike(taxPayer.FirstName, firstNameSearchPattern)
+                where string.IsNullOrEmpty(lastname) || EF.Functions.ILike(taxPayer.LastName, lastNameSearchPattern)
+                where string.IsNullOrEmpty(address) || EF.Functions.ILike(property.AddressLine1.Replace(" ", ""), addressSearchPattern)
+                where string.IsNullOrEmpty(postcode) || EF.Functions.ILike(property.PostCode.Replace(" ", ""), postcodeSearchPattern)
+                select new TaxPayerInformation
+                {
+                    AccountRef = taxPayer.AccountRef,
+                    Uprn = property.Uprn,
+                    FirstName = taxPayer.FirstName,
+                    LastName = taxPayer.LastName,
+                    TaxPayerAddress = new Address
+                    {
+                        AddressLine1 = property.AddressLine1,
+                        AddressLine2 = property.AddressLine2,
+                        AddressLine3 = property.AddressLine3,
+                        AddressLine4 = property.AddressLine4,
+                        PostCode = property.PostCode
+                    }.ToDomain(),
+                    EmailList = new List<string>
+                        {
+                            email.EmailAddress
```

Comment:
I could be wrong, but I think this just adds one element to the email list. I think we might need to add functionality to add more than one email to the email list.

Reply:
I'm looking at the infrastructure for email, and I'm only seeing a single email column in there, compared with PhoneNumber where there are columns for phone number 1 - 4, which is why I only had the single email present. 

Adapting this code to look a bit more like the PhoneNumber list is very possible, but I'm not sure if that is reflective of what we get back from the database.

Reply:
We wouldn't change the infrastructure here but we would want to consider that more than one record in the table could have the same `reference_Id` but have a different email address saved to the email column. So I think, we would want to grab these email addresses as well.